### PR TITLE
feat: add Tauri v2 support

### PR DESCRIPTION
**Pull request by lucasfernog-crabnebula**
This pull request changes the devtools Tauri plugin to use the Tauri v2 alpha. It has been tested on macOS, Linux, iOS and Android.

It is still a draft because we should decide how to maintain both versions of the plugin. Probably the v2 should live in a separate repo/branch.
Also it needs a Tauri PR / release before merging.

Needs #171

### CODE REVIEW

**Comment by lucasfernog-crabnebula** [comment]
@CrabNejonas this one is ready for review (I'm marking it as draft because we want to move the v2 plugin out of this repo - or keep it on a separate branch - and merge the rest).

**Review by CrabNejonas — APPROVED** [review]
As we said the frontend work needs to be merged into v1.x too (right now the 1.x frontend can't connect to the 2.x frontend) but otherwise this looks good to me 👍

File: web-client/src/lib/span/get-span-children.ts
[the remark this answers is not in the document]
Comment (lucasfernog-crabnebula) [reply web-client/src/lib/span/get-span-children.ts:1]: this is only checked once when the span is incoming so idk if it's worth the overhead of storing it instead.. i'll leave it to you guys later

**Comment by johann-crabnebula** [comment]
This is UX issue:

I noticed that the Cert breaks the console display since it overflows:

chrome_SUA6ST2eIT

**Review by johann-crabnebula — CHANGES_REQUESTED** [review]
Noticed that the sources are not working for me on Android:

chrome_SlSVMTcB5U

```sh
01-15 18:23:39.935  6985  7048 I RustStdoutStderr: 2024-01-15T18:23:39.935495Z ERROR devtools::server: List Entries failed with error Io(Os { code: 13, kind: PermissionDenied, message: "Permission denied" })
```

**Review by johann-crabnebula — APPROVED** [review]
I will approve the frontend, so we can move forward.

**Pull request merged**